addons/mail/static/src/core/common/composer.js

```diff
@@ -803,13 +811,29 @@ export class Composer extends Component {
         }
     }
 
-    restoreContent() {
+    async restoreContent() {
         const composer = toRaw(this.props.composer);
         try {
             const config = JSON.parse(browser.localStorage.getItem(composer.localId));
             if (config.text) {
                 composer.emailAddSignature = config.emailAddSignature;
                 composer.text = config.text;
+                await this.suggestion.suggestionService.fetchPartners(
+                    config.mentionedPartners.filter(
+                        (m) => !this.store.Persona.get(m)
+                    ).map((m) => m.id),
+                    this.thread
+                );
+                for (const mentionedPartner of config.mentionedPartners) {
+                    this.props.composer.mentionedPartners.push(
+                        this.store.Persona.get(mentionedPartner)
+                    );
+                }
+                const channelsPromises = config.mentionedChannels.map((c) => this.store.Thread.getOrFetch(c));
+                const channels = await Promise.all(channelsPromises);
+                for (const channel of channels) {
+                    this.props.composer.mentionedChannels.push(channel);
+                }
```

Comment:
This needs to filter channels that are undefined (no access/not existing).

And now that I think about it, it should also use the actual suggestion flow, rather than `getOrFetch` as there are _conditions_ to make a channel selectable, not just its existence (`searchChannelSuggestions` in particular, or a new function to have common logic, as sorting can be omitted here).